packages/medusa/src/api-v2/admin/price-lists/middlewares.ts

```diff
@@ -27,4 +35,14 @@ export const adminPriceListsRoutesMiddlewares: MiddlewareRoute[] = [
       ),
     ],
   },
+  {
+    method: ["GET"],
+    matcher: "/admin/price-lists/:id/products",
```

Comment:
Looking at this with a fresh mind now 😄 

Shouldn't this work a bit differently, where you call `/admin/products` with a filter on the price lists instead? It feels strange to me to ask for the products under a price list, as if the price lists own the product entity. I think the relationship is the other way around.